backend/apollo/elections/views.py

```diff
@@ -99,6 +101,34 @@ def get_permissions(self) -> List[BasePermission]:
         return [perm() for perm in permission_classes]
 
 
-class VoteViewSet(mixins.CreateModelMixin, viewsets.GenericViewSet):
+class VoteViewSet(viewsets.GenericViewSet):
     queryset = Vote.objects.all()
     serializer_class = VoteSerializer
+
+    def create(self, request):
+        try:
+            answer_id_ciphertext = request.data["answer"]
+            election_id = request.data["election"]
+        except KeyError as e:
+            return Response(f"Key {e} not found", status=status.HTTP_400_BAD_REQUEST)
+
+        election = get_object_or_404(Election, pk=election_id)
+
+        try:
+            encoder = nacl.encoding.Base64Encoder()
+            election_sk = nacl.public.PrivateKey(
+                election.secret_key.encode("ascii"), encoder
+            )
+            sealed_box = nacl.public.SealedBox(election_sk)
+            answer_id = sealed_box.decrypt(answer_id_ciphertext, encoder).decode()
+        except nacl.exceptions.CryptoError:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        answer = get_object_or_404(Answer, pk=answer_id)
+        if answer.question.election.id != election_id:
+            return Response("Wrong election id", status=status.HTTP_400_BAD_REQUEST)
```

Comment:
So everything above this line could be in `validate()` in the serializer, and in the last line of this method you'd just set
```python
def validate(attrs):
    attrs = super().validate(attrs)
    ...  # all the crypto stuff
    attrs["answer"] = get_object_or_404(Answer, pk=answer_id)
    return attrs
```